include/effectmanager.h

```diff
@@ -96,6 +95,61 @@ class EffectManager : public IJSONSerializable
         }
     }
 
+    void LoadJSONAndMissingEffects(const JsonArrayConst& effectsArray)
+    {
+        std::set<int> loadedEffectNumbers;
+
+        // Create effects from JSON objects, using the respective factories in g_EffectFactories
+        auto& jsonFactories = g_EffectFactories.GetJSONFactories();
+
+        for (auto effectObject : effectsArray)
+        {
+            int effectNumber = effectObject[PTY_EFFECTNR];
+            auto factoryEntry = jsonFactories.find(effectNumber);
+
+            if (factoryEntry == jsonFactories.end())
+                continue;
+
+            auto pEffect = factoryEntry->second(effectObject);
+
+            if (pEffect != nullptr)
+            {
+                _vEffects.push_back(pEffect);
+                loadedEffectNumbers.insert(effectNumber);
+            }
+        }
+
+        // Now add missing effects from the default factory list
+        auto &defaultFactories = g_EffectFactories.GetDefaultFactories();
+
+        // We iterate manually, so we can use where we are as the starting point for a later inner loop
+        for (auto iter = defaultFactories.begin(); iter != defaultFactories.end(); iter++)

```

Comment:
auto & again I think?

Reply:
I tried that first, but then the code refused to compile.